### PR TITLE
add forbidden slug option

### PR DESCRIPTION
**Pull request by hotsaucejake**
This PR allows the ability to specify forbidden slugs that should NOT be generated.  My use case was to avoid route collision.  I have models with soft deletes and setup routes to handle various tasks like:

```
model/{model}
model/trashed // index of all soft deleted models
```

What if the user creates a model that generates one of these slugs?  They won't be able access their model or perform these tasks.  There are probably more use cases than this one.

```php
class Page extends Model
{
        HasSlug;

        public function getSlugOptions(): SlugOptions
        {
            return parent::getSlugOptions()
                ->forbiddenSlugs(['trashed', 'annihilate', 'restore']);
        }
}
```

### CODE REVIEW

**Comment by hotsaucejake** [comment]
This will also account for unspecified forbidden slugs or an empty array:

```php
public function getSlugOptions(): SlugOptions
{
        return parent::getSlugOptions()
            ->forbiddenSlugs();
}
```

```php
public function getSlugOptions(): SlugOptions
{
        return parent::getSlugOptions()
            ->forbiddenSlugs([]);
}
```

**Comment by hotsaucejake** [comment]
Had to include the composer file because of failed checks

**Comment by hotsaucejake** [comment]
@freekmurze there is a fix to the checks now - it was a composer update to allow for the plugin in the config section

File: src/HasSlug.php [hunk 1]
```diff
@@ -130,7 +134,19 @@ protected function makeSlugUnique(string $slug): string
         $originalSlug = $slug;
         $i = 1;
 
-        while ($this->otherRecordExistsWithSlug($slug) || $slug === '') {
+        while ($this->otherRecordExistsWithSlug($slug) || $slug === '' || in_array($slug, $this->slugOptions->forbiddenSlugs)) {
```
Comment (freekmurze) [inline src/HasSlug.php:137]: This is unreadable. Could you extract this to a function with a good name? Also try to avoid `||` in there.

**Review by freekmurze — CHANGES_REQUESTED** [review]

**Comment by freekmurze** [comment]
Thanks, but after thinking things through, I'm going to pass on this because I want to keep the package as simple as possible.

People that need this behaviour can override the `generateNonUniqueSlug()` method.

**Comment by hotsaucejake** [comment]
@freekmurze this is the same comment left on this pull request: https://github.com/spatie/laravel-sluggable/pull/242 that you just closed.  Is it for the same reason?

This particular pull request allows a user to define a list of slugs that should NOT be generated.  

I'm not sure how overriding the the `generateNonUniqueSlug()` method applies here or one would go about not allowing certain slugs to be generated.  Only the `HasTranslatableSlug` trait contains this method.

**Comment by freekmurze** [comment]
I think you could adapt [this method](https://github.com/spatie/laravel-sluggable/blob/7435973fadf5d2b77ba2b1d2c4b2d62cd5162c8a/src/HasSlug.php#L90-L98) to see if `$this->getSlugSourceString()` returns a forbidden slug source.

**Comment by fabiomlferreira** [comment]
@freekmurze I believe it would be beneficial to revisit the potential addition of a reserved words feature in the sluggable plugin. It's common practice in many sluggable solutions to provide an easy mechanism for adding reserved words, and I think integrating this feature would significantly enhance the utility and user experience of our package. It's a feature that could provide substantial value to many users.

**Pull request closed without merging**